homeassistant/components/systemmonitor/sensor.py

```diff
@@ -76,17 +77,164 @@
+            _LOGGER.warning(
+                "Failed to load process with ID: %s, old name: %s",
+                err.pid,
+                err.name,
+            )
+    return state
+
+
+def get_network(entity: SystemMonitorSensor[dataT]) -> float | None:
+    """Return network in and out."""
+    counters = cast(dict[str, snetio], entity.coordinator.data)
+    if entity.argument in counters:
+        counter = counters[entity.argument][IO_COUNTER[entity.entity_description.key]]
+        return round(counter / 1024**2, 1)
+    return None
+
+
+def get_packets(entity: SystemMonitorSensor[dataT]) -> float | None:
+    """Return packets in and out."""
+    return get_network(entity)
+
+
+def get_throughput(entity: SystemMonitorSensor[dataT]) -> float | None:
+    """Return network throughput in and out."""
+    counters = cast(dict[str, snetio], entity.coordinator.data)
+    if entity.argument in counters:
+        counter = counters[entity.argument][IO_COUNTER[entity.entity_description.key]]
+        now = dt_util.utcnow()
+        if entity.value and entity.value < counter:
+            entity.update_time = now
+            entity.value = counter
+            return round(
+                (counter - entity.value)
+                / 1000**2
+                / (now - (entity.update_time or now)).total_seconds(),
+                3,
+            )
+        entity.update_time = now
+        entity.value = counter
+    return None
+
+
+def get_ip_address(entity: SystemMonitorSensor[dataT]) -> str | None:
+    """Return network ip address."""
+    addresses = cast(dict[str, list[snicaddr]], entity.coordinator.data)
+    if entity.argument in addresses:
+        for addr in addresses[entity.argument]:
+            if addr.family == IF_ADDRS_FAMILY[entity.entity_description.key]:
+                return addr.address
+    return None
+
+
+def get_last_boot(entity: SystemMonitorSensor[dataT]) -> datetime:
+    """Return last boot."""
+    return cast(datetime, entity.coordinator.data)
+
+
+def get_load_1m(entity: SystemMonitorSensor[dataT]) -> float:
+    """Return load 1m."""
+    return round(cast(tuple[float, float, float], entity.coordinator.data)[0], 2)
+
+
+def get_load_5m(entity: SystemMonitorSensor[dataT]) -> float:
+    """Return load 5m."""
+    return round(cast(tuple[float, float, float], entity.coordinator.data)[1], 2)
+
+
+def get_load_15m(entity: SystemMonitorSensor[dataT]) -> float:
+    """Return load 15m."""
+    return round(cast(tuple[float, float, float], entity.coordinator.data)[2], 2)
+
+
 @dataclass(frozen=True)
-class SysMonitorSensorEntityDescriptionMixin(Generic[_dataT]):
+class SysMonitorSensorEntityDescriptionMixin(Generic[dataT]):
     """Mixin for System Monitor sensor entities."""
 
-    coordinator: type[MonitorCoordinator]
-    value_fn: Callable[[_dataT], float | str | datetime | None]
+    value_fn: Callable[[SystemMonitorSensor[dataT]], StateType | datetime]
 
 
 @dataclass(frozen=True)
```

Comment:
We don't need a mixin if we set `kw_only=True`.
```suggestion
@dataclass(frozen=True, kw_only=True)
```